### PR TITLE
Runcard update

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
This PR implements some methods useful for the qiboteam/qibocal#279:

1. `dump` method that saves the runcard in a file
2.  `update` method that modifies the runcard given a dictionary as input.  

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Comment by stavros11** [comment]
@Edoardo-Pedicillo @DavidSarlle are we going to merge this PR as discussed? If yes, could you or @rodolfocarobene fix the conflicts here? They were caused by merging #379 to main.

**Comment by rodolfocarobene** [comment]
Now it should be possible to merge without problems.
Github has having some issue with tests, but they are all passing locally (I tested qpu tests just on the rfsoc)

**Comment by Edoardo-Pedicillo** [comment]
Thank @rodolfocarobene, tests are passing now (https://github.com/qiboteam/qibolab/actions/runs/4945290957)

File: src/qibolab/platforms/abstract.py
Comment (rodolfocarobene) [inline src/qibolab/platforms/abstract.py:178]: Why are we converting from GHz to Hz? I thought we were using Hz everywhere

**Review by rodolfocarobene — COMMENTED** [review]
In general it looks good to me, could you maybe add some documentation?

File: src/qibolab/platforms/abstract.py
Comment (stavros11) [reply src/qibolab/platforms/abstract.py:178]: Good point. @andrea-pasquale @Edoardo-Pedicillo does the autocalibration return the value in GHz?

Comment (andrea-pasquale) [reply src/qibolab/platforms/abstract.py:178]: The fitting is now performed considering GHz and not Hz to improve the stability, also in the plotting we show GHz and not BHz as it was happening before which was not ideal. This is why we are currently using `GHz`.
We could also convert it back to Hz just for the runcard update mechanism. Let me know what you think.

Comment (stavros11) [reply src/qibolab/platforms/abstract.py:178]: Either solution is fine with me, as long as it is documented that `platform.update` expects the frequencies in GHz.

**Review by stavros11 — APPROVED** [review]
If this has been tested to work with qblox and tii1q_b1, I would suggest to merge it asap so that we can unlock other PRs that are waiting for the autocalibration, particularly the new execution options and results and the Zurich driver.

**Review by rodolfocarobene — APPROVED** [review]

**Comment by stavros11** [comment]
@andrea-pasquale @Edoardo-Pedicillo I restarted the workflow to see if it passes (it had an issue uploading the coverage). When you are ready with this, feel free to merge.

**Comment by andrea-pasquale** [comment]
> @andrea-pasquale @Edoardo-Pedicillo I restarted the workflow to see if it passes (it had an issue uploading the coverage). When you are ready with this, feel free to merge.

Thanks @stavros11. We would like to add documentation, as you suggested, and tests. This should not take long.
@Edoardo-Pedicillo will take care of this.

**Review by andrea-pasquale — APPROVED** [review]
Thanks @Edoardo-Pedicillo.
We will take care of coverage after the merge of #387.

**Pull request merged**
